capstone-backend/app.js

```diff
@@ -17,6 +17,105 @@ app.get("/", (req, res) => {
   res.send("It works!")
 })
 
+app.get('/getUser', async (req, res) => {
+  const query = new Parse.Query("User");
+      try {
+      const userdata = await query.find();
+      res.send(userdata)
+      } catch (error) {
+        res.send(error)
+      }
+  })
+
+  function checkUserInterests(searchVal, interestInfo, usersInfo){
+    const userInfoJson = usersInfo.toJSON();
+    for (let skill of interestInfo.skills) {
+      const jsonSkill = skill.toJSON(); 
+      if (jsonSkill.name?.toLowerCase().includes(searchVal)) {
+        return true;
+      }
+    }
+    for (let company of interestInfo.companies) {
+      const jsonCompany = company.toJSON();
+      if (jsonCompany.name?.toLowerCase().includes(searchVal)) {
+        return true;
+       
+      }
+    }
+    for (let language of interestInfo.languages) {
```

Comment:
Just FYI, these for of logic can be rewritten with Array.some
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
```
const languageMatch = interestInfo.languages.some(
	language => {
		const jsonLanguage = language.toJSON();
		return jsonLanguage.name?.toLowerCase().includes(searchVal);
	}
);
if (languageMatch) {
	return true;
}
```

And if you want to reduce the number of lines:
```
if (interestInfo.languages.some(
	language => language.toJSON().name?.toLowerCase().includes(searchVal);
)) {
	return true;
}
```